app/controllers/build_estimates_controller.rb

```diff
@@ -1,61 +1,70 @@
 class BuildEstimatesController < ApplicationController
   include Wicked::Wizard
-  include BuildEstimatesHelper
+  include StepsHelper
 
-  steps(*ALL_ESTIMATE_STEPS)
+  steps(*STEPS_WITH_PROPERTY)
 
-  MONTHLY_INCOME_ATTRIBUTES = MonthlyIncomeForm::INCOME_ATTRIBUTES + [:monthly_incomes]
-
-  def new
-    session[session_key] = {}
-    redirect_to wizard_path(steps.first)
-  end
+  HANDLER_CLASSES = {
+    applicant: Flow::ApplicantHandler,
+    monthly_income: Flow::IncomeHandler,
+    property: Flow::PropertyHandler,
+    vehicle: Flow::VehicleHandler,
+    assets: Flow::AssetHandler,
+    summary: Flow::SummaryHandler,
+    outgoings: Flow::OutgoingsHandler,
+    property_entry: Flow::PropertyEntryHandler,
+  }.freeze
 
   def show
-    case step
-    when :intro
-      @estimate = IntroForm.new session[session_key]
-    when :monthly_income
-      @estimate = MonthlyIncomeForm.new session[session_key].slice(*MONTHLY_INCOME_ATTRIBUTES)
-    when :property
-      @intro_form = IntroForm.new session[session_key]
-    end
+    handler = HANDLER_CLASSES[step]
+    @form = if handler
+              handler.model(session_data)
+            else
+              cfe_connection.api_result(estimate_id)
+            end
+    @estimate = load_estimate
     render_wizard
   end
 
   def update
-    case step
-    when :intro
-      @estimate = IntroForm.new(intro_params)
-      if @estimate.valid?
-        session[session_key] = @estimate.attributes
-        redirect_to wizard_path next_step_for(@estimate, step)
-      else
-        render_wizard
-      end
-    when :monthly_income
-      @estimate = MonthlyIncomeForm.new(monthly_income_params)
-      if @estimate.valid?
-        session[session_key].merge(@estimate.attributes)
+    handler = HANDLER_CLASSES.fetch(step)
+    @form = handler.form(params)
+
+    if @form.valid?
+      if step == :summary
+        estimate = load_estimate
+        cfe_connection.create_applicant estimate_id,
+                                        date_of_birth: estimate.over_60 ? 61.years.ago.to_date : 59.years.ago.to_date,
```

Comment:
thanks, I have been trying to remove all this logic out of the controller so that it just looks after flow (we will probably have sub-flows soon or something) so I've managed to move this into SummaryHandler#save_data where it seems to belong